driver/fvGFS/fv_nggps_diag.F90

```diff
@@ -913,6 +964,61 @@ subroutine fv_nggps_tavg(Atm, Time_step_atmos,avg_max_length,zvir)
     endif
    endif
 
+   if ( id_o3_ave > 0 .and. id_no_ave >0 &
+       .and. id_no2_ave > 0 .and. id_pm25_ave >0 ) then
+
+        allocate ( o3(isco:ieco,jsco:jeco,npzo) )
```

Comment:
To save memory, I'd suggest not adding new 3D temporary arrays, instead, 1) adding atm%q, trace_idx, and units as arguments to average_tracer_hy1. 2) call the average_tracer_hy1 for those new variables o2_idx, no_idx, etc.